std/json.d

```diff
@@ -278,7 +314,7 @@ struct JSONValue
         return type == JSON_TYPE.NULL;
     }
 
-    private void assign(T)(T arg)
+    private void assign(T)(T arg) @safe
```

Comment:
Better error messages this way. If I edit `assign` to make it not safe, I want to see the offending line rather than 3-5 other places saying `cannot call method 'assign' from @safe code` and no further enlightenment.